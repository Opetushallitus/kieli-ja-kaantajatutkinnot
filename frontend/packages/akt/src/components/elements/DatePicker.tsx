import Stack from '@mui/material/Stack';
import TextField from '@mui/material/TextField';
import { Dayjs } from 'dayjs';

import { DateUtils } from 'utils/date';

export interface DatePickerProps {
  value?: string;
  setValue: (value: string) => void;
  label: string;
  disabled?: boolean;
  minDate?: Dayjs;
  maxDate?: Dayjs;
  dataTestId?: string;
}

export const DatePicker = ({
  value,
  setValue,
  label,
  disabled = false,
  minDate,
  maxDate,
  dataTestId,
}: DatePickerProps): JSX.Element => {
  const MIN_DATE = '1900-01-01';
  const MAX_DATE = '2100-01-01';
  const dayjs = DateUtils.dayjs();

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setValue(e.target.value);
  };

  return (
    <Stack spacing={3}>
      <TextField
        data-testid={dataTestId ? `${dataTestId}__date-picker` : 'date-picker'}
        disabled={disabled}
        label={label}
        type="date"
        onChange={handleChange}
        value={value}
        InputLabelProps={{
          shrink: true,
        }}
        inputProps={{
          min: DateUtils.formatOptionalDate(minDate ?? dayjs(MIN_DATE)),
          max: DateUtils.formatOptionalDate(maxDate ?? dayjs(MAX_DATE)),
        }}
      />
    </Stack>
  );
};
