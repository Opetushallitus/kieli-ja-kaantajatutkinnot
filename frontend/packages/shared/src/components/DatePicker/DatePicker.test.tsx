import renderer from 'react-test-renderer';

import { DatePicker } from './DatePicker';

describe('DatePicker', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <DatePicker
          label="test label"
          value="2022-04-21"
          setValue={jest.fn()}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
