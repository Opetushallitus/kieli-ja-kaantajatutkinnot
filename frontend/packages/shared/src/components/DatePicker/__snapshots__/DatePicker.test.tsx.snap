// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker should render correctly 1`] = `
<div
  className="css-1lwbda4-MuiStack-root"
>
  <div
    className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    data-testid="date-picker"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={true}
      htmlFor=":r0:"
      id=":r0:-label"
    >
      test label
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id=":r0:"
        max="2100-01-01"
        min="1900-01-01"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="date"
        value="2022-04-21"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-1in441m"
        >
          <span>
            test label
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
